lib/jira/commands/install.rb

```diff
@@ -15,35 +15,52 @@ class Install < Base
 
       def run
         io.say('Please enter your JIRA information.')
-        inifile[:global] = params
+        inifile[:global] = base_params
+        inifile.write # Do this now because cookie authentication uses api calls
+
+        inifile.delete_section("cookie") if inifile.has_section?("cookie")
+        case authentication
+        when "basic"
+          inifile[:global][:password] = password
+        when "token"
+          inifile[:global][:token] = token
+        when "cookie"
```

Comment:
Not really.
I assume the user would try an command and Jira would respond with 'errorMessages' that the cookie session is no longer valid.
The best we can do is have the user re-authenticate for a cookie session.

Response from a cookie session is:
{
    "session": {
        "name": "JSESSIONID",
        "value": "12345678901234567890"
    },
    "loginInfo": {
        "failedLoginCount": 10,
        "loginCount": 127,
        "lastFailedLoginTime": "2016-01-27T07:30:51.349+0000",
        "previousLoginTime": "2016-01-27T07:30:51.349+0000"
    }
}